apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudTrace/TraceHeaderContext.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Text.RegularExpressions;
+using System.Web;
+
+namespace Google.Devtools.AspNet
+{
+    /// <summary>
+    /// Context from the Cloud Trace Header.
+    /// </summary>
+    /// 
+    /// <remarks>
+    /// A trace can be forced by passing information along in the trace header ("X-Cloud-Trace-Context").
+    /// The trace id, parent span id and whether or not to trace can be set.
+    /// See: https://cloud.google.com/trace/docs/faq#how_do_i_force_a_request_to_be_traced
+    /// </remarks>
+    internal sealed class TraceHeaderContext
+    {
+        // An TraceHeaderContext representing no information or invalid information from a header.
+        private static readonly TraceHeaderContext InvalidTraceHeaderContext =
+            new TraceHeaderContext(null, null, false);
+
+        // The trace header.
+        public static readonly string TraceHeader = "X-Cloud-Trace-Context";
+
+        // A regex to match the trace header. 
+        // - ([A-Fa-f0-9]{32}): The trace id, a 32 character hex value.
+        // - ([0-9]+): The span id, a 64 bit integer.
+        // - (o=([0-3])): The trace mask, 1-3 denote it should be traced.
+        public static readonly Regex TraceHeaderRegex = new Regex(@"([A-Fa-f0-9]{32})/([0-9]+);(o=([0-3]))?$");
```

Comment:
Start the regex with '^' so it matches to the start of the string. Also, specify RegexOptions.Compiled as the 2nd parameter for faster matches.